packages/server/src/cache/client.ts

```diff
@@ -0,0 +1,43 @@
+import path from 'path'
+import fs from 'fs'
+import Redis, { Redis as IORedis } from 'ioredis'
+import config from '../config/server'
+import { scriptRateLimit } from './timeouts'
+import { scriptSetLeaderboard, scriptGetRange, scriptSetGraph, scriptGetGraph } from './leaderboard'
+
+const creds = config.database.redis
+
+type RedisClient = IORedis & {
+  rctfRateLimit: scriptRateLimit,
+  rctfSetLeaderboard: scriptSetLeaderboard,
+  rctfGetRange: scriptGetRange,
+  rctfSetGraph: scriptSetGraph,
+  rctfGetGraph: scriptGetGraph
+}
+
+let client: RedisClient
+
+if (typeof creds === 'string') {
+  client = new Redis(creds, {
+    dropBufferSupport: true
+  }) as RedisClient
+} else {
+  client = new Redis({
+    host: creds.host,
+    port: creds.port,
+    password: creds.password,
+    db: creds.database,
+    dropBufferSupport: true
+  }) as RedisClient
```

Comment:
Destructuring `creds` and then constructing the options object seems much more verbose than the current code, especially because it would require slightly special handling of `db` / `database`.